src/Domain/TagsCharacteresticSpeeds.hpp

```diff
@@ -32,7 +32,7 @@ struct CharSpeedCompute : CharSpeedsComputeTag::base, db::ComputeTag {
   template <typename... Ts, typename T, size_t NumberOfCharSpeeds>
   static void function(
       const gsl::not_null<std::array<T, NumberOfCharSpeeds>*> result,
-      const boost::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+      const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
```

Comment:
please change `static_cast<bool>(grid_velocity)` below to `grid_velocity.has_value()`

Reply:
Agreed about removing the static_casts. Do you prefer `has_value()` to the bool operator? Using the bool operator, these static casts become, e.g., `if (grid_velocity) {...}` which is nicely concise.

Reply:
I prefer the more explicit `has_value` as I find it much easier to read and understand